contentcuration/contentcuration/static/js/edit_channel/channel_list/views/ChannelStar.vue

```diff
@@ -0,0 +1,36 @@
+<template>
+    <Star
+      :starred="channel.STARRED"
+      @starred="addStar(channel)"
+      @unstarred="removeStar(channel)"
+    />
+</template>
+
+<script>
+
+import { mapActions } from 'vuex';
+import Star from 'edit_channel/sharedComponents/Star.vue';
+
+export default {
+  name: 'ChannelStar',
+  components: {
+    Star
+  },
+  props: {
+    channel: {
+      type: Object,
```

Comment:
This is related to my comment above - in general, passing Objects as props can be a bit of a pain, because it's harder to validate the data that is coming in. (although in this case, this kind of just defers the problem to the data store where we are storing the channel data).